flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -373,8 +464,21 @@ public Builder() {
 		public Builder field(String name, DataType dataType) {
 			Preconditions.checkNotNull(name);
 			Preconditions.checkNotNull(dataType);
-			fieldNames.add(name);
-			fieldDataTypes.add(dataType);
+			columns.add(TableColumn.of(name, dataType));
+			return this;
+		}
+
+		/**
+		 * Add a field with name and computation expression.
+		 * This column would be a computed column.
+		 *
+		 * <p>The call order of this method determines the order of fields in the schema.
+		 */
+		public Builder field(String name, DataType dataType, String expression) {
```

Comment:
Please add a detailed description for parameters, especially for `expression` string. 
The `expression` string will be persisted to catalogs, so it must be expanded and qualified, otherwise, the `expression` may not work when restored from catalog. You can refer the javadoc of watermark expression string.